cmake/scripts/webos/Install.cmake

```diff
@@ -50,6 +50,16 @@ file(WRITE ${CMAKE_BINARY_DIR}/install.cmake "
   if(EXISTS ${BIN_ADDONS_DIR})
     file(INSTALL ${BIN_ADDONS_DIR} DESTINATION ${APP_PACKAGE_DIR})
   endif()
+
+  file(GLOB_RECURSE LIBRARIES \"${APP_PACKAGE_DIR}/lib/python3/site-packages/Cryptodome/*x86_64-linux-gnu.so\")
```

Comment:
My only concern is the hardcoded x86_64. Is there a chance people are going to build on different host architectures for this (eg arm/aarch64 based host)?

Yes it works for things based on our existing CI platform, but are you going to get issues crop up from people who are self compiling?

Reply:
Yes, that's also a concern I have. I have adapted it to match for the host triplet from the filename. However, it is quite ugly. Is there no way to get the host triplet from some existing cmake variable?

Reply:
atm we only have the target triplet, but you could add the build host triplet to Toolchain.cmake

Reply:
Not sure if maybe https://cmake.org/cmake/help/latest/variable/CMAKE_HOST_SYSTEM_PROCESSOR.html#variable:CMAKE_HOST_SYSTEM_PROCESSOR will give you the correct cpu info from host. Won't be the triplet, but might give you the arch

Reply:
Well, there's a target triplet but it differs from what the python modules expect. `CMAKE_HOST_SYSTEM_PROCESSOR` is empty

Reply:
Just propagate it through the tool chain file then

Reply:
```cmake
set(HOST_VENDOR "@host_vendor@")
set(HOST_OS "@host_os@")
```

Resolves into after configure

```cmake
set(HOST_VENDOR "webos")
set(HOST_OS "linux-gnueabi")
```

Reply:
`@build` gives `x86_64-pc-linux-gnu` which is close but it has an additional `-pc-` in it.

Reply:
It seems python is doing it's own thing of deriving the host and target triplets. Not sure if we can do better than the regex matching

Reply:
the only concern is the cpu (eg x86_64). Id say its safe to assume the linux-gnu side of the triplet (i know the toolchain cant be built on macos, because i tried this a couple months back), so just propagate @use_buildcpu@ through to use to regex the x86_64

Reply:
It actually can be built on mac with some tricks. There's a precompiled one [here for intel macs](https://github.com/webosbrew/native-toolchain). I wanted to take a look at building the toolchain for arm macs too as I'm planning swapping my macbook for a arm one

Reply:
i wouldnt over think it at this point in time. At this point you're almost better handling it when PIL is actually built and rename the libs before you patch the loaders.

Reply:
I am wondering why we do not use something like 
```bash
export SETUPTOOLS_EXT_SUFFIX=.cpython-311-arm-linux-gnueabi.so
``` when building. That environment variable allows to completely control the suffix of the compiled libraries.

Reply:
You don't know what you don't know ;)

Reply:
Alright, let's drop all of these packaging hacks and just set `export SETUPTOOLS_EXT_SUFFIX=.so` during building.

Let's just drop the platform completely from the library name which is fine because pycryptodome falls back to a platformless library name as last resort:

`error Contents: Cannot load native module 'Cryptodome.Cipher._raw_ecb': Not found '_raw_ecb.cpython-311-arm-linux-gnueabi.so', Not found '_raw_ecb.abi3.so', Not found '_raw_ecb.so'` 

Possibly this fix also applies to Android and might be able to replace some of the patches that we currently apply

Reply:
so with the SETUPTOOLS_EXT_SUFFIX, you probably dont need the following any longer?https://github.com/xbmc/xbmc/blob/b29d4d26b2a602659748b051308fa7b9f3f439b8/tools/depends/target/pythonmodule-pil/Makefile#L92-L94

i set the ext_suffix for macos, and checked the loaders, and they correctly match the lib in the _imaging*.py loaders

Reply:
And the android stuff will still be needed. All libs are put in a separate folder in the apk (ie not in the site-package folder other platforms can deal with), and all android libs must start with lib prefix for the lib name. unless you know some magical prefix env var :P